src/Testcontainers.Nats/NatsBuilder.cs

```diff
@@ -0,0 +1,115 @@
+using DotNet.Testcontainers;
+
+namespace Testcontainers.Nats;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class NatsBuilder : ContainerBuilder<NatsBuilder, NatsContainer, NatsConfiguration>
+{
+    public const string NatsImage = "nats:2.9";
+
+    public const ushort ClientPort = 4222;
+    public const ushort RoutingPort = 6222;
+    public const ushort MonitoringPort = 8222;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NatsBuilder" /> class.
+    /// </summary>
+    public NatsBuilder()
+        : this(new NatsConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NatsBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private NatsBuilder(NatsConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override NatsConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the Nats Server password.
+    /// </summary>
+    /// <param name="password">The Nats Server password.</param>
+    /// <returns>A configured instance of <see cref="NatsBuilder" />.</returns>
+    public NatsBuilder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new NatsConfiguration(password: password))
+            .WithCommand("--pass", password);
+    }
+
+    /// <summary>
+    /// Sets the Nats Server username.
+    /// </summary>
+    /// <param name="username">The Nats Server username.</param>
+    /// <returns>A configured instance of <see cref="NatsBuilder" />.</returns>
+    public NatsBuilder WithUsername(string username)
+    {
+        return Merge(DockerResourceConfiguration, new NatsConfiguration(username: username))
+            .WithCommand("--user", username);
+    }
+
+    /// <inheritdoc />
+    public override NatsContainer Build()
+    {
+        Validate();
+        return new NatsContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    
+    /// <inheritdoc />
+    protected override void Validate()
+    {
+        base.Validate();
+        
+        if (DockerResourceConfiguration.Password != null || DockerResourceConfiguration.Username != null)
+        {
+            _ = Guard.Argument(DockerResourceConfiguration.Username, nameof(DockerResourceConfiguration.Username))
+                .NotNull();
+
+            _ = Guard.Argument(DockerResourceConfiguration.Password, nameof(DockerResourceConfiguration.Password))
+                .NotNull();
+        }
```

Comment:
QQ: What do you think about a similar approach we use in the MongoDB module and set `string.empty` as default values for `username` and `password` and add these validation checks:

https://github.com/testcontainers/testcontainers-dotnet/blob/0c6588ed28b2bc14c19373e029f3421949440247/src/Testcontainers.MongoDb/MongoDbBuilder.cs#L92-L99

It makes the subsequent null-checks unnecessary. Note, the `UriBuilder` returns a Uri without `:` if the `username` and `password` is empty, see: https://github.com/testcontainers/testcontainers-dotnet/pull/983#issuecomment-1700376825.